scripts/zones/West_Ronfaure/npcs/Vilatroire.lua

```diff
@@ -5,29 +5,222 @@
+            end
+
+            if (questIntroToTeamwork == QUEST_ACCEPTED) then
+                -- https://www.bg-wiki.com/bg/Introduction_to_Teamwork
+                if (partySameNationCount == partySizeRequirement) then
+                    -- nation requirements met
+                    player:setCharVar("introToTmwrk_pass", 1);
+                    player:updateEvent(15, 1);
+                else
+                    -- not met
+                    player:updateEvent(3);
+                end
+            elseif (questIntermediateTeamwork == QUEST_ACCEPTED) then
+                -- https://www.bg-wiki.com/bg/Intermediate_Teamwork
+                if (partySameRaceCount == partySizeRequirement) then
+                    -- race requirements met
+                    player:setCharVar("intermedTmwrk_pass", 1);
+                    player:updateEvent(15, 2);
+                else
+                    -- not met
+                    player:updateEvent(4);
+                end
+            elseif (questAdvancedTeamwork == QUEST_ACCEPTED) then
+                -- https://www.bg-wiki.com/bg/Advanced_Teamwork
+                if (partySameJobCount == partySameJobCount) then
+                    -- race requirements met
+                    player:setCharVar("advTmwrk_pass", 1);
+                    player:updateEvent(15, 3);
+                else
+                    -- not met
+                    -- UPDATE ME --
+                    player:updateEvent(5);
+                end
+            end
+        else
+            -- need more party members
+            player:updateEvent(1);
+        end
+    end
 end
 
 function onEventFinish(player, csid, option)
-end
+    -- csid 129 is the event for when they have selected ready/not ready
+    
+    if (csid == 129 and option == 0) then
+        local questIntroToTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK);
+        local questIntermediateTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTERMEDIATE_TEAMWORK);
+        local questAdvancedTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.ADVANCED_TEAMWORK);
+
+            if (questIntroToTeamwork == QUEST_ACCEPTED and player:getCharVar("introToTmwrk_pass") == 1) then
+                -- check their inventory
+                if (player:getFreeSlotsCount() == 0) then
+                    player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, 13442);
+                else
+                    player:completeQuest(SANDORIA,tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK);
+                    player:addFame(SANDORIA, 80); -- unsure of actual fame value
+                    player:addTitle(tpz.title.THIRDRATE_ORGANIZER);
+                    player:addItem(13442); -- shell ring
+                    player:messageSpecial(ID.text.ITEM_OBTAINED, 13442); -- shell ring
+                    player:setCharVar("introToTmwrk_pass", 0) -- Delete charVar from memory
+                end
```

Comment:
```lua
        npcUtil.completeQuest(player, SANDORIA, INTRODUCTION_TO_TEAMWORK, {
            item = { 13342 },
            fame = 80, -- fame defaults to 30 if not set
            title = tpz.title.THIRDRATE_ORGANIZER,
            var = {"introToTmwrk_pass"}
        })
```
` npcUtil.completeQuest` will do all of this for you. Also check out our [Lua Style Guide](https://github.com/project-topaz/topaz/blob/release/CONTRIBUTING.md). Haven't really dug in to the logic but it looks like it works, though you could condense some other stuff too, like
```lua
if (player:getRace() == tpz.race.HUME_M and member:getRace() == tpz.race.HUME_M or member:getRace() == tpz.race.HUME_F) then
```
could be
```lua
if (player:getRace() == tpz.race.HUME_M or player:getRace() == tpz.race.HUME_F) and 
    (member:getRace() == tpz.race.HUME_M or member:getRace() == tpz.race.HUME_F) then

Reply:
Changing to the npcUtil seemed to make my server crash... this is what I have.
```lua
if questIntroToTeamwork == QUEST_ACCEPTED and player:getCharVar("introToTmwrk_pass") == 1 then
                -- check their inventory
                npcUtil.completeQuest(player, SANDORIA, INTRODUCTION_TO_TEAMWORK, {
                  item = {13442},
                  fame = 80,
                  title = tpz.title.THIRDRATE_ORGANIZER,
                  var = {"introToTmwrk_pass"}
                })
```

Reply:
Sorry, my syntax may have been off. I updated my comment above, should work now. Don't forget to `require("scripts/globals/npc_util")`